enhancements/installer/connected-onprem-assisted-installer.md

```diff
@@ -0,0 +1,236 @@
+
+## Motivation
+
+Many users will not want a service that can provision hosts inside their
+network to be run by a third party outside of their network. Reasons can
+include trust, control, and reproducibility. Thus those users will prefer to
+run the service on-premise.
+
+### Goals
+
+* Users can run the entire assisted install workflow with software that runs inside their network.
+
+### Non-Goals
+
+* Disconnected and air-gapped environments are not included in this enhancement.
+* Adding Nodes on day 2 will be addressed separately.
+* Utilizing a local container registry for mirrored content will be addressed in a future enhancement.
+
+## Proposal
+
+A user will visit cloud.redhat.com to download a live ISO with which they can
+run the assisted installer on-premise. The RHCOS-based ISO will include:
+
+* the user's pull secret
+* a systemd unit file that runs all containers necessary to run the assisted installer
+
+Similar to downloading the current installer, the download will default to the
+latest available version of the assisted installer, RHCOS and OpenShift.
+
+### User Stories [optional]
+
+#### Story 1
+
+I want to install OpenShift while running the assisted installer and the entire
+installation workflow on infrastructure that I control. I do not want a third
+party cloud service to control the provisioning of my hosts from across the
+internet.
+
+#### Story 2
+
+I want to automate use of the assisted installer and expect reproducible
+results. Utilizing a cloud service does not provide sufficient control or
+guarantees of reproducibility.
+
+#### Story 3
+
+I want to test the assisted installer prior to production use and expect
+reproducible results. Utilizing a cloud service does not provide sufficient
+control or guarantees of reproducibility.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Single Use on Day 1 Only
+
+The on-premise assisted installer will run once per cluster. As a live ISO, it
+will not utilize any persistent storage. Upon completion of a cluster
+installation, the live ISO will be shut down and all state will vanish.
+
+#### Creating the Agent ISO
+
+The assisted installer generates a live ISO that can be booted on hosts and then
+used to provision them. It runs the Agent that communicates with the assisted
+installer service. The ISO needs to include:
+
+* a URL to where the assisted installer is running
+* a CA certificate so it can trust the assisted installer
+
+The assistant needs local access to a base RHCOS ISO in order to generate the
+Agent ISO. It will then run the following command, passing ignition either via
+`stdin` or as an additional argument.
+
+```
+coreos-installer iso embed -o /<output-path>/agent.iso /path/to/base/rhcos.iso
+```
+
+To facilitate local access, the base ISO image will be included inside the
+assisted installer's container image.
+
+#### S3
+
+The cloud implementation stores each generated ISO in an S3-like service. This
```

Comment:
The version for test environments uses a service with an S3-compatible API.  The cloud.redhat.com deployment will use the actual AWS S3 service.

Reply:
Would you suggest a change to the language here? I figured that whether production uses actual S3 or any other S3-compatible service is an implementation detail that doesn't impact this proposal.

Reply:
Maybe "in S3" rather than "in an S3-like service"